### PR TITLE
Refactor(torii-core): Optimize and clean sql statements

### PR DESCRIPTION
**Pull request by gianalarcon**
Refactor fn register_model(), and fn set_entity() to get rid of extra SQL query statements.

Solution
Use RETURNING clause to avoid unnecessary SQL statements

### CODE REVIEW

**Comment by broody** [comment]
Thanks! there are some sizable changes coming down the pipe (https://github.com/dojoengine/dojo/pull/916), I think we could wait for this one to land first. Also we're gonna be moving to `block_id` instead of `created_at` at some point.

**Comment by broody** [comment]
Reopening for @gianalarcon now that nested model is implemented

**Comment by gianalarcon** [comment]
Ready for review!

File: crates/torii/core/src/sql.rs [hunk 1]
```diff
@@ -153,24 +147,20 @@ impl Sql {
...
+        let insert_entities = format!(
             "INSERT INTO entities (id, keys, model_names, event_id) VALUES ('{}', '{}', '{}', \
              '{}') ON CONFLICT(id) DO UPDATE SET model_names=excluded.model_names, \
-             updated_at=CURRENT_TIMESTAMP, event_id=excluded.event_id",
+             updated_at=CURRENT_TIMESTAMP, event_id=excluded.event_id RETURNING created_at",
             entity_id, keys_str, model_names, event_id
-        ));
+        );
+        // execute first to get created_at
+        let query_result: SqliteRow = sqlx::query(&insert_entities).fetch_one(&self.pool).await?;
```
Comment (tarrencev) [inline crates/torii/core/src/sql.rs:157]: Can you use `query_as` here?

**Review by tarrencev — APPROVED** [review]
minor suggestion otherwise lgtm!

File: crates/torii/core/src/sql.rs (hunk 1 above)
Comment (tarrencev) [reply crates/torii/core/src/sql.rs:157]: Merging. Feel free to follow up with the change

**Pull request merged**